coq/utils.ml

```diff
@@ -45,47 +45,14 @@ let to_range ~lines (p : Loc.t) : Lang.Range.t =
 
 let to_orange ~lines = Option.map (to_range ~lines)
 
-(* Separation of parsing and execution made upstream API hard to use for us
-   :/ *)
-let pmeasure (measure, print) fn =
-  match measure fn () with
-  | Ok _ as r -> Feedback.msg_notice @@ print r
-  | Error (exn, _) as r ->
-    Feedback.msg_notice @@ print r;
-    Exninfo.iraise exn
-
-let with_fail fn =
-  try
-    fn ();
-    CErrors.user_err (Pp.str "The command has not failed!")
-  with exn when CErrors.noncritical exn ->
-    let exn = Exninfo.capture exn in
-    let msg = CErrors.iprint exn in
-    Feedback.msg_notice ?loc:None
-      Pp.(str "The command has indeed failed with message:" ++ fnl () ++ msg)
-
-let with_ctrl ctrl ~st ~fn =
-  let st = State.to_coq st in
-  match ctrl with
-  | Vernacexpr.ControlTime ->
-    pmeasure System.(measure_duration, fmt_transaction_result) fn
-  | Vernacexpr.ControlInstructions ->
-    pmeasure System.(count_instructions, fmt_instructions_result) fn
-  | Vernacexpr.ControlTimeout n -> (
-    match Control.timeout (float_of_int n) fn () with
-    | None -> Exninfo.iraise (Exninfo.capture CErrors.Timeout)
-    | Some x -> x)
-  (* fail and succeed *)
-  | Vernacexpr.ControlFail ->
-    with_fail fn;
-    Vernacstate.Interp.invalidate_cache ();
-    Vernacstate.unfreeze_full_state st
-  | Vernacexpr.ControlSucceed ->
-    fn ();
-    Vernacstate.Interp.invalidate_cache ();
-    Vernacstate.unfreeze_full_state st
-  (* Unsupported by coq-lsp, maybe deprecate upstream *)
-  | Vernacexpr.ControlRedirect _ -> fn ()
```

Comment:
Why is redirect unsupported? Should I filter them out before sending the control flags to VernacControl?

Reply:
I guess no real reason, mainly because it is a kind of security hole, and indeed, you'd like the document manager or build system to be notified that a command is going to write a file.

But IMHO things are fine as they are, if people use `Redirect` in `coq-lsp` they will get files written in their FS, their choice I guess.